### PR TITLE
fgsl 1.0.0 (new formula)

### PR DESCRIPTION
**Pull request by tschoonj**
Hi guys,

I created a formula for FGSL, the Fortran bindings to the GNU Scientific Library. I figured homebrew-science would make a nice home for this one.

Everything seems to work fine, but I will of course make whatever changes you would like me to. This is my first attempt at making a Homebrew Formula, so please go easy on me :-)

By the way someone was asking for this Formula just a couple of days ago at Homebrew/homebrew#35327

### CODE REVIEW

**Comment by dpo** [comment]
Thanks! This is nice. Could you please squash your commits into one and push to your branch?

**Comment by tschoonj** [comment]
Done!

**Comment by dpo** [comment]
@mikemcquaid The bot says
````
Error Message

failed: brew audit fgsl --strict
Stacktrace

        Error: 1 problems in 1 formulae
fgsl:
 * A `test do` test block should be added
````
Why is it really necessary to have a `test do` block? Like numerous other `science` formulas, this ones runs a `make check` (which passes) and it doesn't even contain an executable (it's a library). In fact, `gsl.rb` doesn't contain a `test do` block either.

**Comment by mikemcquaid** [comment]
It's good practice for formulae. `make check` does not validate it installed or bottled correctly.

**Comment by tschoonj** [comment]
Could you recommend me how to write an appropriate `test do` block for this formula? It provides merely a .dylib file, a Fortran module and some documentation...

**Comment by mikemcquaid** [comment]
See `tinyxml.rb` for an example of a library with a good test.

**Comment by tschoonj** [comment]
Interesting situation. The following works fine

```ruby
test do
    ENV.fortran
    system ENV.fc, "#{share}/examples/fgsl/fft.f90", "-L#{lib}", "-lfgsl" ,
    "-I#{include}/fgsl", "-o", "test"
    system "./test"
end
```
but `brew audit`complains:

```
Use `depends_on :fortran` instead of `ENV.fortran`
```

However if make that change I end up with:

```
Error: fgsl: failed
undefined method `depends_on' for #<Fgsl:0x007ff9b41f7c40>
/usr/local/Library/Taps/tschoonj/homebrew-science/fgsl.rb:22:in `block in <class:Fgsl>'
/usr/local/Library/Homebrew/formula.rb:609:in `block in run_test'
/usr/local/Library/Homebrew/extend/fileutils.rb:21:in `mktemp'
/usr/local/Library/Homebrew/formula.rb:607:in `run_test'
/usr/local/Library/Homebrew/cmd/test.rb:59:in `block (2 levels) in test'
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/timeout.rb:66:in `timeout'
/usr/local/Library/Homebrew/cmd/test.rb:58:in `block in test'
/usr/local/Library/Homebrew/cmd/test.rb:37:in `each'
/usr/local/Library/Homebrew/cmd/test.rb:37:in `test'
/usr/local/Library/brew.rb:136:in `<main>'
```

Removing `ENV.fortran` is not an option because then `ENV.fc` evaluates to `nil`...

Any advice?

**Comment by dpo** [comment]
You shouldn't need to say `ENV.fortran` if you've already said `depends_on :fortran` in the class body.

**Comment by tschoonj** [comment]
With ENV.fortran in there I get:

```
Testing fgsl
==> Using Homebrew-provided fortran compiler.
This may be changed by setting the FC environment variable.
==> /usr/local/bin/gfortran /usr/local/Cellar/fgsl/1.0.0/share/examples/fgsl/fft.f90 -L/usr/local/Cellar/fgsl/1.0.0/lib -lfgsl -I/us
==> ./test
```

Without:

```
Testing fgsl
==> /usr/local/Cellar/fgsl/1.0.0/share/examples/fgsl/fft.f90 -L/usr/local/Cellar/fgsl/1.0.0/lib -lfgsl -I/usr/local/Cellar/fgsl/1.0.
Error: fgsl: failed
no implicit conversion of nil into String
/usr/local/Library/Homebrew/formula.rb:652:in `basename'
/usr/local/Library/Homebrew/formula.rb:652:in `system'
/usr/local/Library/Taps/tschoonj/homebrew-science/fgsl.rb:22:in `block in <class:Fgsl>'
/usr/local/Library/Homebrew/formula.rb:609:in `block in run_test'
/usr/local/Library/Homebrew/extend/fileutils.rb:21:in `mktemp'
/usr/local/Library/Homebrew/formula.rb:607:in `run_test'
/usr/local/Library/Homebrew/cmd/test.rb:59:in `block (2 levels) in test'
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/timeout.rb:66:in `timeout'
/usr/local/Library/Homebrew/cmd/test.rb:58:in `block in test'
/usr/local/Library/Homebrew/cmd/test.rb:37:in `each'
/usr/local/Library/Homebrew/cmd/test.rb:37:in `test'
/usr/local/Library/brew.rb:136:in `<main>'
```

Weird

File: fgsl.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    system "./configure", "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}"
+    system "make"
+    system "make", "check" if build.with? "check"
+    system "make", "install"
+  end
+  test do
+    ENV.fortran
+    system ENV.fc, "#{prefix}/share/examples/fgsl/fft.f90", "-L#{prefix}/lib", "-lfgsl" ,
```
Comment (dpo) [inline fgsl.rb]: Replace `#{prefix}/share` with `#{share}`, and `#{prefix}/lib` with `#{lib}`.

File: fgsl.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    "-I#{prefix}/include/fgsl", "-o", "test"
```
Comment (dpo) [inline fgsl.rb]: `#{prefix}/include` -> `#{include}`.

**Comment by iml** [comment]
Hmm the dependencies may not be accessed by `test`. If this is intentional, `ENV.fortran` is indeed needed in the test method and `audit` should only check the install. cc @jacknagel 

We could also just ignore the warning, since it's a niche case, or use `which gfortran`, which is what `ENV.fortran` sets `FC` to in most cases (and the bot).

**Comment by jacknagel** [comment]
The warning is a false positive (the audit check is just a textual comparison to the string "ENV.fortran". It can be safely ignored.

**Comment by dpo** [comment]
But it's not a warning, it's an error, isn't it? As far as I can tell, bottles weren't built.

**Comment by mikemcquaid** [comment]
They are built anyway. If you can submit a PR for a better `audit` check we'd likely accept it.

**Comment by dpo** [comment]
Sorry, I'm a bit confused because there's no `pr-1606` tag on the bot's fork and `brew pull --bottle` doesn't reveal any `bottle` block.  I see the `*.bottle.rb` files on the control panel, but how do I pull those?

**Comment by mikemcquaid** [comment]
http://bot.brew.sh/job/Homebrew%20Science%20Pull%20Requests/598/downstreambuildview/
http://bot.brew.sh/job/Homebrew%20Bottles/12373/console

The bottle job needs retried. Kicking it off now.

**Comment by dpo** [comment]
Thanks @mikemcquaid!

**Comment by tschoonj** [comment]
Thanks to everybody!

**Pull request closed without merging**